### PR TITLE
Log: Implement compile error event

### PR DESCRIPTION
**Pull request by WingToh**
Test the following:
-Open a new project/solution
-Close a new project/solution
-Try to compile with a compile error

### CODE REVIEW

**Review by JustNireon — COMMENTED** [review]
LGTM up till the last commit since i've commited this myself. This should be reviewed by someone else.

File: PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs [hunk 1]
```diff
@@ -85,11 +79,14 @@ public static async Task SubscribeEventHandlersAsync()
...
         public static async Task UnsubscribeEventHandlersAsync()
         {
-
             await _package.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationToken);
             _dte.Events.BuildEvents.OnBuildDone -= OnCompileDone;
+            _dte.Events.SolutionEvents.BeforeClosing -= OnProjectClose;
         }
 
+
+        #region MyRegion
```
Comment (ovda96) [inline PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs:88]: I don't think that you want this as the region title, right?

File: PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs [hunk 2]
```diff
@@ -187,28 +186,128 @@ internal static void OnSessionEnd()
...
+        /// </summary>
+        private static void OnProjectOpen()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            Projects projects = _dte.Solution.Projects;
+
+            // Distinguish whether only a csproj file was opened initially, or a solution file.
+            // For a csproj, _dte.Solution.FullName returns an empty string and the projects inside
+            // have to be iterated (which is only 1 project, the csproj file).
+            string nametest = _dte.Solution.FullName;
```
Comment (ovda96) [inline PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs:217]: According to naming convention: nameTest

**Review by ovda96 — CHANGES_REQUESTED** [review]

**Comment by ovda96** [comment]
@WingToh Please, also adapt the PR-name to reflect the commit conventions.

File: PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs [hunk 3]
```diff
@@ -75,8 +65,12 @@ public static class SubscribeEvents
         public static async Task SubscribeEventHandlersAsync()
         {
             await _package.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationToken);
+            _dteSolution = _dte.Solution;
             // Code that interacts with UI elements goes here
             _dte.Events.BuildEvents.OnBuildDone += OnCompileDone;
+            _dte.Events.SolutionEvents.Opened += OnProjectOpen;
+            _dte.Events.SolutionEvents.BeforeClosing += OnProjectClose;
+            _dte.Events.DebuggerEvents.OnEnterDesignMode += OnRunProgram; //Not triggering...
```
Comment (ovda96) [inline PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs:73]: Maybe add `TODO` before the comment?

**Review by ovda96 — CHANGES_REQUESTED** [review]

**Review by ovda96 — APPROVED** [review]

**Pull request merged**